src/identity/app.rs

```diff
@@ -61,6 +61,14 @@ impl FullId {
     pub fn public_id(&self) -> &PublicId {
         &self.public_id
     }
+
+    /// Constructs a `FullId` for a given BLS secret key.
+    pub fn with_keys(bls_sk: BlsSecretKey, owner: PublicKey) -> Self {
```

Comment:
Is this c'tor only required for testing?  Same question for `ClientFullId::with_bls_key()`.

Reply:
I needed them so that I could construct respsective `FullId`s for already existing clients and apps. This is then assigned to the mock routing struct as I mentioned before.